src/xo-app/new-vm/index.js

```diff
@@ -500,6 +502,23 @@ export default class NewVm extends BaseComponent {
     }
   }
   _onChangeSshKeys = keys => this._setState({ sshKeys: map(keys, key => key.id) })
+  _getOnChangeIps = index => ips => {
```

Comment:
I really dislike this factories!

Usually it means you should have created a dedicated component.